experimental/packages/opentelemetry-sdk-node/src/start.ts

```diff
@@ -54,13 +84,141 @@ export function startNodeSDK(sdkOptions: SDKOptions): {
   );
+  const resource = setupResource(config, sdkOptions);
+  const loggerProvider = setupLoggerProvider(config, sdkOptions, resource);
 
   const shutdownFn = async () => {
     const promises: Promise<unknown>[] = [];
+    if (loggerProvider) {
+      promises.push(loggerProvider.shutdown());
+    }
     await Promise.all(promises);
   };
   return { shutdown: shutdownFn };
 }
 const NOOP_SDK = {
   shutdown: async () => {},
 };
+
+function setupResource(
+  config: ConfigurationModel,
+  sdkOptions: SDKOptions
+): Resource {
+  let resource: Resource = sdkOptions.resource ?? defaultResource();
+  const autoDetectResources = sdkOptions.autoDetectResources ?? true;
+  let resourceDetectors: ResourceDetector[];
+
+  if (!autoDetectResources) {
+    resourceDetectors = [];
+  } else if (sdkOptions.resourceDetectors != null) {
+    resourceDetectors = sdkOptions.resourceDetectors;
+  } else if (config.node_resource_detectors) {
+    resourceDetectors = getResourceDetectorsFromConfiguration(config);
+  } else {
+    resourceDetectors = [
+      envDetector,
+      processDetector,
+      hostDetector,
+      osDetector,
+      serviceInstanceIdDetector,
+    ];
+  }
+
+  if (autoDetectResources) {
+    const internalConfig: ResourceDetectionConfig = {
+      detectors: resourceDetectors,
+    };
+
+    resource = resource.merge(detectResources(internalConfig));
+  }
+
+  resource =
+    sdkOptions.serviceName === undefined
+      ? resource
+      : resource.merge(
+          resourceFromAttributes({
+            [ATTR_SERVICE_NAME]: sdkOptions.serviceName,
+          })
+        );
+
+  return resource;
+}
+
+function setupLoggerProvider(
+  config: ConfigurationModel,
+  sdkOptions: SDKOptions,
+  resource: Resource | undefined
+): LoggerProvider | undefined {
+  const logProcessors =
+    sdkOptions.logRecordProcessors ??
+    getLogRecordProcessorsFromConfiguration(config);
+
+  if (logProcessors) {
+    const loggerProvider = new LoggerProvider({
+      resource: resource,
+      processors: logProcessors,
+    });
+
+    logs.setGlobalLoggerProvider(loggerProvider);
+    return loggerProvider;
+  }
+  return undefined;
+}
+
+function getLogRecordExporter(
+  exporter: LogRecordExporterModel
+): LogRecordExporter {
+  if (exporter.otlp_http) {
+    if (exporter.otlp_http.encoding === 'json') {
+      return new OTLPHttpLogExporter({
+        compression:
+          exporter.otlp_http.compression === 'gzip'
+            ? CompressionAlgorithm.GZIP
+            : CompressionAlgorithm.NONE,
+      });
+    }
+    if (exporter.otlp_http.encoding === 'protobuf') {
+      return new OTLPProtoLogExporter();
+    }
+    diag.warn(`Unsupported OTLP logs protocol. Using http/protobuf.`);
+    return new OTLPProtoLogExporter();
+  } else if (exporter.otlp_grpc) {
+    return new OTLPGrpcLogExporter();
+  } else if (exporter.console) {
+    return new ConsoleLogRecordExporter();
+  }
+  diag.warn(`Unsupported Exporter value. Using OTLP http/protobuf.`);
```

Comment:
🤔  If someone uses a config file but neglects to specify a valid exporter (`otlp_http`, `otlp_grpc`, `otlp_file/development`(soon), `console`)... should we default an exporter for them or should we consider it a noop because it's invalid? It's not clear to me.

Reply:
Sorry on the delay in coming back to this. In re-reviewing the spec for [required and null properties](https://github.com/open-telemetry/opentelemetry-configuration/blob/main/CONTRIBUTING.md#required-and-null-properties), my understanding is that if a field is required it must be a valid entry to be used and otherwise it is invalid. 
> When a property is required, the key must be included in the object or the configuration is invalid. 

I'll take a look at the go and java implementations to see how they handle it